src/mtcp/mtcp_restart.c

```diff
@@ -850,7 +850,10 @@ unmap_memory_areas_and_restore_vdso(RestoreInfo *rinfo)
       mtcp_abort();
     }
     MTCP_ASSERT(vvar == vvarStart);
-    mtcp_memcpy(vvarStart, rinfo->vvarStart, vvarEnd - vvarStart);
+    // On i386, only the first 4096 bytes are readable. Reading beyond that
+    // results in a bus error.
```

Comment:
Let's also add here:
` *     // This issue arose due to a change in the Linux kernel pproximately in version 4.0`
The rationale is that we will then know that this doesn't matter in some future case where a customer/user has not upgraded from an older kernel for the sake of robustness.